src/main.cpp

```diff
@@ -6802,108 +6802,121 @@ bool static ProcessMessage(const CChainParams& chainparams, CNode* pfrom, string
             return false;
         }
 
-        int64_t nTime;
-        CAddress addrMe;
-        CAddress addrFrom;
-        uint64_t nNonce = 1;
-        std::string strSubVer;
-        std::string cleanSubVer;
-        uint64_t nServices;
-        vRecv >> pfrom->nVersion >> nServices >> nTime >> addrMe;
-        pfrom->nServices = nServices;
-        if (pfrom->nVersion < MIN_PEER_PROTO_VERSION)
-        {
-            // disconnect from peers older than this proto version
-            LogPrintf("peer=%d using obsolete version %i; disconnecting\n", pfrom->id, pfrom->nVersion);
-            pfrom->PushMessage("reject", strCommand, REJECT_OBSOLETE,
-                               strprintf("Version must be %d or greater", MIN_PEER_PROTO_VERSION));
+        // Initialize all temporaries defaults
+        int32_t nVersion{0};
+        uint64_t nServices{0};
+        int64_t nTime{0};
+        CAddress addrMe{};
+        CAddress addrFrom{};
+        uint64_t nNonce{1U};
+        std::string strSubVer{};
+        std::string cleanSubVer{};
+        int32_t nStartingHeight{0};
+        bool fRelay{true};
+
+        // Consume all available payload
+        // This block might throw an IO exception. If that's the case then
+        // nothing in pfrom's state will be changed.
+        vRecv >> nVersion >> nServices >> nTime >> addrMe;
+        if (nVersion >= 106) {
+            if (vRecv.in_avail() < 39) {
+                // Be strict about size of addr_from + nonce + user_agent + start_height
+                pfrom->PushMessage("reject", strCommand, REJECT_MALFORMED, string("Error in message size"));
+                pfrom->fDisconnect = true;
+                return false;
+            }
+            vRecv >> addrFrom >> nNonce >> LIMITED_STRING(strSubVer, MAX_SUBVERSION_LENGTH) >> nStartingHeight;
+            cleanSubVer = SanitizeString(strSubVer, SAFE_CHARS_SUBVERSION);
+            if (nVersion >= 70001) {
```

Comment:
Where does this version number come from? It doesn't appear in Zcash source, and all of the Bitcoin version numbers had 100000 added to them. But I don't know why we would be checking `nVersion >= 170001` either. See `src/version.h` for all of the version numbers that should be relevant to Zcash.

Maybe this was backported from some other Bitcoin fork? Which one?